src/transport.cpp

```diff
@@ -490,6 +520,47 @@ namespace quicr {
         }
     }
 
+    void Transport::SubscribeAnnounces(TransportConnId conn_id,
+                                       const std::shared_ptr<SubscribeAnnouncesHandler>& handler)
+    {
+        // Generate track alias
+        auto ns_prefix = handler->GetTrackNamespacePrefix();
+
+        SPDLOG_LOGGER_INFO(logger_, "SubscribeAnnounces track conn_id: {0}", conn_id);
+
+        std::unique_lock<std::mutex> lock(state_mutex_);
+
+        auto conn_it = connections_.find(conn_id);
+        if (conn_it == connections_.end()) {
+            SPDLOG_LOGGER_ERROR(logger_, "SubscribeAnnounces conn_id: {0} does not exist.", conn_id);
+            return;
+        }
+
+        auto it = conn_it->second.announce_subscriptions.find(ns_prefix);
+        if (it != conn_it->second.announce_subscriptions.end()) {
+            // do nothing
+            return;
+        }
+
+        conn_it->second.announce_subscriptions[ns_prefix] = std::move(handler);
```

Comment:
`it` already is `conn_it->second.announce_subscriptions[ns_prefix]`, we don't need to look it up again. In map this is effectively an O(1) operation though, so maybe it doesn't matter.